OutOfSchool/OutOfSchool.WebApi/Services/OrganizationService.cs

```diff
@@ -49,80 +49,68 @@ public async Task<OrganizationDTO> Create(OrganizationDTO dto)
+                : "Successfully got all records from the Organization table.");
+
+            return organizations.Select(organization => organization.ToModel()).ToList();
         }
 
         /// <inheritdoc/>
         public async Task<OrganizationDTO> GetById(long id)
         {
+            logger.Information("Process of getting Organization by id started.");
+
             var organization = await repository.GetById(id).ConfigureAwait(false);
+
             if (organization == null)
             {
-                throw new ArgumentException("Incorrect Id!", nameof(id));
+                throw new ArgumentOutOfRangeException(id.ToString(),
+                    "The id cannot be greater number of table entities.");
             }
 
+            logger.Information($"Successfully got a Organization with id = {id}.");
+
             return organization.ToModel();
         }
 
         /// <inheritdoc/>
         public async Task<OrganizationDTO> Update(OrganizationDTO dto)
         {
-            if (dto == null)
-            {
-                throw new ArgumentNullException(nameof(dto), "Organization was null.");
-            }
-
-            if (dto.EDRPOU.Length == 0)
-            {
-                throw new ArgumentException("EDRPOU code is empty", nameof(dto));
-            }
-
-            if (dto.INPP.Length == 0)
-            {
-                throw new ArgumentException("INPP code is empty", nameof(dto));
-            }
-
-            if (dto.MFO.Length == 0)
-            {
-                throw new ArgumentException("MFO code is empty", nameof(dto));
-            }
-
-            if (dto.Title.Length == 0)
-            {
-                throw new ArgumentException("Title is empty", nameof(dto));
-            }
-
-            if (dto.Description.Length == 0)
-            {
-                throw new ArgumentException("Description is empty", nameof(dto));
-            }
-
             try
             {
                 var organization = await repository.Update(dto.ToDomain()).ConfigureAwait(false);
-              
+
+                logger.Information("Updating successfully finished.");
+
                 return organization.ToModel();
             }
-            catch (Exception ex)
+            catch (DbUpdateConcurrencyException)
             {
-                throw new Exception($"{nameof(dto)} could not be updated: {ex.Message}");
+                logger.Error("Updating failed. There is no Organization in the Db with such an id.");
+                throw;
             }
         }
-        
+
         /// <inheritdoc/>
         public async Task Delete(long id)
         {
+            var dtoToDelete = new Organization() { Id = id };
+
             try
             {
-                await repository
-                    .Delete(await repository.GetById(id).ConfigureAwait(false))
-                    .ConfigureAwait(false);
+                await repository.Delete(dtoToDelete).ConfigureAwait(false);
```

Comment:
small note
dtoToDelete could be named entity or organization
code will read better then: repository.Delete(entity)